apps/examples/testcase/le_tc/kernel/tc_task.c

```diff
@@ -282,6 +349,16 @@ static void tc_task_task_restart(void)
 	int ret_chk;
 	unsigned int remain;
 	g_callback = 0;
+
+	/* Check for NULL pid parameter  */
+
+	ret_chk = task_restart(0);
+	TC_ASSERT_EQ("task_restart", ret_chk, ERROR);
+	TC_ASSERT_EQ("task_restart", errno, ENOSYS);
+	TC_ASSERT_EQ("task_restart", g_callback, 0);
```

Comment:
1. Previous logic is better. Calling ```task_restart``` is unnecessary test without any launched task.
```
	pid = task_create("tc_exit", SCHED_PRIORITY_MAX - 1, 1024, restart_task, (char * const *)NULL);
	TC_ASSERT_GT("task_create", pid, 0);

	g_callback = 0;
	ret_chk = task_restart(0);
	TC_ASSERT_EQ("task_restart", ret_chk, ERROR);
	TC_ASSERT_EQ("task_restart", errno, ENOSYS);
	TC_ASSERT_EQ("task_restart", g_callback, 0);
```

2. As I left, you need to initialize checking value, g_callback.
  As you add new testcase before testing below, if someone adds new tc before below, we can't guarantee g_callback is zero.

Reply:
ok understood why re-initialize it,  will make these changes